modules/RTC/ortc/RTCPeerConnection.js

```diff
@@ -0,0 +1,532 @@
+    }
+
+    /**
+     */
+    getLocalStreams() {
+        return Array.from(this._localStreams);
+    }
+
+    /**
+     */
+    getRemoteStreams() {
+        return Array.from(this._remoteStreams);
+    }
+
+    /**
+     */
+    getStats() {
+        // TODO
+    }
+
+    /**
+     *
+     */
+    _createIceGatherer(pcConfig) {
+        const iceGatherOptions = {
+            gatherPolicy: pcConfig.iceTransportPolicy || 'all',
+            iceServers: pcConfig.iceServers || []
+        };
+        const iceGatherer = new RTCIceGatherer(iceGatherOptions);
+
+        // NOTE: Not implemented by Edge.
+        iceGatherer.onstatechange = () => {
+            logger.debug(
+                `iceGatherer "statechange" event, state:${iceGatherer.state}`);
+
+            this._updateAndEmitIceGatheringStateChange(iceGatherer.state);
+        };
+
+        iceGatherer.onlocalcandidate = ev => {
+            let candidate = ev.candidate;
+
+            // NOTE: Not implemented by Edge.
+            const complete = ev.complete;
+
+            logger.debug(
+                'iceGatherer "localcandidate" event, candidate:', candidate);
+
+            // NOTE: Instead of null candidate or complete:true, Edge
+            // signals end of gathering with an empty candidate object.
+            if (complete
+                || !candidate
+                || Object.keys(candidate).length === 0) {
+
+                candidate = null;
+
+                this._updateAndEmitIceGatheringStateChange('complete');
+                this._emitIceCandidate(null);
+            } else {
+                this._emitIceCandidate(candidate);
+            }
+        };
+
+        iceGatherer.onerror = ev => {
+            const errorCode = ev.errorCode;
+            const errorText = ev.errorText;
+
+            logger.error(
+                `iceGatherer "error" event, errorCode:${errorCode}, `
+                + `errorText:${errorText}`);
+        };
+
+        // NOTE: Not supported by Edge, which starts gathering automatically.
+        try {
+            iceGatherer.gather();
+        } catch (error) {
+            logger.warn(`iceGatherer.gather() failed:${error}`);
+        }
+
+        return iceGatherer;
+    }
+
+    /**
+     *
+     */
+    _createIceTransport(iceGatherer) {
+        const iceTransport = new RTCIceTransport(iceGatherer);
+
+        // NOTE: Not implemented by Edge.
+        iceTransport.onstatechange = () => {
+            logger.debug(
+                'iceTransport "statechange" event, '
+                + `state:${iceTransport.state}`);
+
+            this._emitIceConnectionStateChange();
+        };
+
+        // NOTE: Not standard, but implemented by Edge.
+        iceTransport.onicestatechange = () => {
+            logger.debug(
+                'iceTransport "icestatechange" event, '
+                + `state:${iceTransport.state}`);
+
+            this._emitIceConnectionStateChange();
+        };
+
+        // TODO: More stuff.
```

Comment:
Can I do that later? Obviously that is related to not yet done stuff (but I do know that something should be done in there).